src/System.Private.ServiceModel/src/System/ServiceModel/Channels/MessageContent.cs

```diff
@@ -180,15 +179,34 @@ protected override Task<Stream> CreateContentReadStreamAsync()
             // WriteMessageAsync might run synchronously and try to write to the stream. ProducerConsumerStream
             // will block on the write until the stream is being read from. The WriteMessageAsync method needs
             // to run on a different thread to prevent a deadlock.
-            _stream = new ProducerConsumerStream();
-            var bufferedStream = new BufferedWriteStream(_stream, WriteBufferSize);
-            Task.Run(() => _messageEncoder.WriteMessageAsync(_message, bufferedStream));
-            return Task.FromResult(_stream);
+            var resultStream = new ProducerConsumerStream();
+            _stream = new BufferedWriteStream(resultStream, WriteBufferSize);
+            Task.Factory.StartNew(async (content) =>
+            {
+                var thisPtr = content as StreamedMessageContent;
+                try
+                {
+                    await _messageEncoder.WriteMessageAsync(thisPtr._message, thisPtr._stream);
```

Comment:
Before and after this change, if WriteMessageAsync throws, the exception will get eaten.  Is that ok?